include/zcbor_print.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZCBOR_PRINT_H__
+#define ZCBOR_PRINT_H__
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef ZCBOR_PRINT_FUNC
+#include <stdio.h>
+#define zcbor_do_print(...) printf(__VA_ARGS__)
+#else
+#define zcbor_do_print(...) ZCBOR_PRINT_FUNC(__VA_ARGS__)
+#endif
+
+#ifdef ZCBOR_VERBOSE
+#define zcbor_trace() (zcbor_do_print("bytes left: %zu, byte: 0x%x, elem_count: 0x%zx, err: %d, %s:%d\n",\
+	(size_t)state->payload_end - (size_t)state->payload, *state->payload, state->elem_count, \
+	state->constant_state ? state->constant_state->error : 0, __FILE__, __LINE__))
+
+#define zcbor_log_assert(expr, ...) \
+do { \
+	zcbor_do_print("ASSERTION \n  \"" #expr \
+		"\"\nfailed at %s:%d with message:\n  ", \
+		__FILE__, __LINE__); \
+	zcbor_do_print(__VA_ARGS__);\
+} while(0)
+#define zcbor_log(...) zcbor_do_print(__VA_ARGS__)
+#else
+#define zcbor_trace() ((void)state)
+#define zcbor_log_assert(...)
+#define zcbor_log(...)
+#endif
+
+#ifdef ZCBOR_ASSERTS
+#define zcbor_assert(expr, ...) \
+do { \
+	if (!(expr)) { \
+		zcbor_log_assert(expr, __VA_ARGS__); \
+		ZCBOR_FAIL(); \
+	} \
+} while(0)
+#define zcbor_assert_state(expr, ...) \
+do { \
+	if (!(expr)) { \
+		zcbor_log_assert(expr, __VA_ARGS__); \
+		ZCBOR_ERR(ZCBOR_ERR_ASSERTION); \
+	} \
+} while(0)
+#else
+#define zcbor_assert(expr, ...)
+#define zcbor_assert_state(expr, ...)
+#endif
+
+__attribute__((used))
+static void zcbor_print_compare_lines(const uint8_t *str1, const uint8_t *str2, uint32_t size)
+{
+	for (uint32_t j = 0; j < size; j++) {
+		zcbor_do_print("%x ", str1[j]);
+	}
+	zcbor_do_print("\r\n");
+	for (uint32_t j = 0; j < size; j++) {
+		zcbor_do_print("%x ", str2[j]);
+	}
+	zcbor_do_print("\r\n");
+	for (uint32_t j = 0; j < size; j++) {
+		zcbor_do_print("%x ", str1[j] != str2[j]);
+	}
+	zcbor_do_print("\r\n");
+	zcbor_do_print("\r\n");
```

Comment:
Does it intentionally print `"\r\n"` twice?

Reply:
Yes, to make it visually separate from the next thing to be printed.